addon/commands/insert-newLi-command.ts

```diff
@@ -0,0 +1,260 @@
+import Command from "./command";
+import Model from "@lblod/ember-rdfa-editor/model/model";
+import ModelSelection from "@lblod/ember-rdfa-editor/model/model-selection";
+import ModelNode from "../model/model-node";
+import ModelElement from "../model/model-element";
+import ModelText from "../model/model-text";
+import {MisbehavedSelectionError, NoParentError, SelectionError} from "@lblod/ember-rdfa-editor/utils/errors";
+import {INVISIBLE_SPACE} from "../model/util/constants";
+
+
+export default class InsertNewLiCommand extends Command {
+  name = "insert-newLi";
+
+  constructor(model: Model) {
+    super(model);
+  }
+
+  canExecute(selection: ModelSelection = this.model.selection): boolean {
+    if (!ModelSelection.isWellBehaved(selection)) {
+      throw new MisbehavedSelectionError();
+    }
+    const commonAncestor = selection.getCommonAncestor().parent;
+    if (commonAncestor?.findAncestor(node => ModelNode.isModelElement(node) && (node.type === 'ul' || node.type === 'ol'), true)) {
+      return true;
+    }
+    return false;
+  }
+
+  execute(): void {
+    const selection = this.model.selection;
+
+    if (!ModelSelection.isWellBehaved(selection)) {
+      throw new MisbehavedSelectionError();
```

Comment:
Currently I think it's the event handlers responsibility to handle these and we currently don't